DiamondLightSource/SynchWeb
# LIMS-1550: Add 'Mark Dispensing' button to plate well view

**JIRA ticket**: [LIMS-1550](https://jira.diamond.ac.uk/browse/LIMS-1550)

**Summary**:

We need a way to store the dispensing location for each well in a plate. This PR adds buttons to mark that point manually, edit it, and delete it, and will also display the currently selected location. The location is stored in the BLSamplePosition table.

**Changes**:
- Add 2 new buttons, 'Mark Dispensing' and (initially hidden) 'Delete'
- Manage the logic between the new buttons and the existing 'Mark Point' and 'Mark Region' buttons
- Insert or update the dispensing position in the database (the delete button updates the x/y/z positions to null)
- Display the dispensing position as a white circle, to distinguish it from subsamples

**To test**:
- Go to proposal nt37104, go to /containers, then click on any plate with subsamples
- Go to a well with subsamples
- Check the 'Mark Point', 'Mark Region' and 'Mark Dispensing' buttons turn to 'Finish' or 'Cancel' buttons when clicked, but only one can be active at once
- Click 'Mark Dispensing' and then click on the well, check a white circle appears to display where you clicked
- Check the white circle remains in place after a refresh, and after zooming in and out
- Check the BLSamplePosition table has a new row or has been updated, with non-null values for posX/Y, and a positionType of 'dispensing'
- Repeat the process, check the position updates on screen and in the database
- Click 'Mark Dispensing' and then click 'Delete', check the white circle disappears, check the posX/Y values are now null
